src/StringTokenizer.h

```diff
@@ -90,9 +90,12 @@ struct MatchResultComparator
             const WordBoundaryMatchResult *wba = static_cast<WordBoundaryMatchResult *>(a.get());
             const WordBoundaryMatchResult *wbb = static_cast<WordBoundaryMatchResult *>(b.get());
 
-            for (size_t i = 0; i < wba->indices.size(); i++)
```

Comment:
what if just `std::min(wba->indices.size(), wbb->indices.size())` ?

Reply:
I considered that, but we're not using the indices at all, so I thought it would be better to use iterators. I can change it back to indices if that's what others would prefer, though.

Reply:
does it looks like std::lexicographical_compare ?
http://en.cppreference.com/w/cpp/algorithm/lexicographical_compare

Reply:
does it looks like std::lexicographical_compare?

Reply:
Wow, I really need to brush up on my standard library algorithms because I didn't even know that exists.

It seems like it's not exactly the same because std::lexicographical_compare returns false if A >= B, so we can't easily determine if A == B (in which case we need to move forward and check the priority). We might still be able to leverage it, though. I'll think about it a little more and get back to you.

Reply:
I don't think we can leverage std::lexicographical_compare without calling it twice because we need to know if the two inputs are equal. We could use std::mismatch and check the returned iterators, but it looks like the behavior is undefined pre-C++14 in cases where the second range is shorter. 

Unless there's another standard library algorithm that I'm missing, I think we might be better off with the current version that uses iterators.